### PR TITLE
Improve parsing for labeled tuple patterns

### PR DESCRIPTION
**Pull request by ccasin**
review: @riaqn

### CODE REVIEW

File: ocaml/parsing/parser.mly [hunk 1]
```diff
@@ -3499,17 +3499,79 @@ pattern_no_exn:
...
+  | mkpat(
+      self COLONCOLON error
         { expecting $loc($3) "pattern" }
     | self BAR pattern
         { Ppat_or($1, $3) }
     | self BAR error
-        { expecting $loc($3) "pattern" }
-  ) { $1 }
+        { expecting $loc($3) "pattern" }  ) { $1 }
+;
```
Comment (riaqn) [inline ocaml/parsing/parser.mly:3519]: some nit pick: seems like we can reduce the diff here?

Comment (riaqn) [inline ocaml/parsing/parser.mly:3519]: some nit pick: seems like we can reduce the diff here?

File: ocaml/parsing/parser.mly [hunk 2]
```diff
@@ -3614,64 +3672,11 @@ simple_delimited_pattern:
...
-      { $1, Open }
-  // Partial patterns are allowed to be length-one
-  | ltuple_component_pat COMMA DOTDOT
-      { [$1], Open }
-
 %inline pattern_semi_list:
   ps = separated_or_terminated_nonempty_list(SEMI, pattern)
     { ps }
 ;
+
```
Comment (riaqn) [inline ocaml/parsing/parser.mly:3679]: extra line

**Review by riaqn — APPROVED** [review]
some minor points, otherwise fine to merge.

**Comment by ccasin** [comment]
@riaqn nice catches, thanks.  Will merge when CI passes.

File: ocaml/parsing/parser.mly [hunk 3]
```diff
@@ -3499,9 +3499,18 @@ pattern_no_exn:
...
+      { let closed, pats = $1 in
+        if    closed = Closed
+           && List.for_all (fun (l,_) -> Option.is_none l) pats
+        then
+          mkpat ~loc:$sloc (Ppat_tuple(List.rev_map snd pats))
+        else
+          ppat_lttuple $sloc (List.rev pats) closed
+      }
+  | mkpat(
+      self COLONCOLON error
```
Comment (riaqn) [inline ocaml/parsing/parser.mly:3513]: Sorry I wasn't clear - I meant that currently there are two `mkpat` (but there was only one) - I think if we keep it that way, we can minimize the diff to only removing `pattern_comma_list` and adding `reversed_labeled_tuple_pattern`.

Comment (ccasin) [reply ocaml/parsing/parser.mly]: Right - I think I see, and agree that is better.  Does it look right now?

Comment (riaqn) [reply ocaml/parsing/parser.mly]: yeah I think it's perfect now - thanks for fullfilling my nits.

**Pull request merged**